data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/prepper/StringPrepper.java

```diff
@@ -57,12 +59,17 @@ public StringPrepper(final Configuration configuration) {
     }
 
     @Override
-    public Collection<Record<String>> execute(final Collection<Record<String>> records) {
-        final Collection<Record<String>> modifiedRecords = new ArrayList<>(records.size());
-        for (Record<String> record : records) {
-            final String recordData = record.getData();
+    public Collection<Record<Event>> execute(final Collection<Record<Event>> records) {
+        final Collection<Record<Event>> modifiedRecords = new ArrayList<>(records.size());
+        for (Record<Event> record : records) {
+            final Event recordEvent = record.getData();
+            final String recordData = recordEvent.toJsonString();
```

Comment:
I don't think this is the correct approach. This will capitalize keys as well, which is incorrect.

I'd think the closest feature parity here is to iterate over every key. For each string-value, make it uppercase.

This processor should probably be obsoleted and replaced with the one being created for #697. But, that is a breaking change and we still need something in 1.3.

Reply:
> I'd think the closest feature parity here is to iterate over every key. For each string-value, make it uppercase.

I agree this might make more sense. Yeah, the Prepper itself is a bit outdated.

Reply:
> 

There seems to be some overlap between the mutate processor and the string_converter proposed here. I think separating out between string mutations and generic mutations makes some sense, but it is possible that doing so could lead to a worse user experience as well.

Reply:
IMO, for the long term, we need to check how much overlap there is between #697 and #508. This Prepper will be transient.

Reply:
I added a comment in #508. We should discuss in that issue how granular we want these different plugins to be. Please read and comment.